server/api/shares.js

```diff
@@ -147,7 +147,26 @@ router.post("shares.update", auth(), async (ctx) => {
   ctx.assertUuid(id, "id is required");
 
   const { user } = ctx.state;
-  const share = await Share.findByPk(id);
+
+  // fetch the share with document and collection.
+  const share = await Share.findByPk(id, {
```

Comment:
Consider moving this into the model as a named scope to keep the controllers clean. Interestingly the default scope includes the user and team too…